modules/auxiliary/gather/zoomeye_search.rb

```diff
@@ -1,178 +1,210 @@
 ##
 # This module requires Metasploit: https://metasploit.com/download
 # Current source: https://github.com/rapid7/metasploit-framework
+# rewrite zoomeye search module 2021/07/14
+#
 ##
 
 
-
 class MetasploitModule < Msf::Auxiliary
   include Msf::Auxiliary::Report
 
-
-  def initialize(info={})
-    super(update_info(info,
-      'Name'        => 'ZoomEye Search',
-      'Description' => %q{
-        The module use the ZoomEye API to search ZoomEye. ZoomEye is a search
-        engine for cyberspace that lets the user find specific network
-        components(ip, services, etc.).
-      },
-      'Author'      => [ 'Nixawk' ],
-      'References'  => [
-        ['URL', 'https://github.com/zoomeye/SDK'],
-        ['URL', 'https://www.zoomeye.org/api/doc'],
-        ['URL', 'https://www.zoomeye.org/help/manual']
-      ],
-      'License'     => MSF_LICENSE
-      ))
-
-      register_options(
-        [
-          OptString.new('USERNAME', [true, 'The ZoomEye username']),
-          OptString.new('PASSWORD', [true, 'The ZoomEye password']),
```

Comment:
Can we retain the username / password functionality and obtain the access code using it if it's provided? I understand not wanting to require the password for security purposes but removing these options would be a breaking change for this module.

I'd suggest at the start of your module requiring that either the username and password are set, or that the API key is set. If credentials are provided then get the API key, then regardless of how you obtained the API key check that it's valid before carrying on.

Reply:
You need to log in again every 12 hours for authentication using username/password, but this problem will not exist if you use API Key.